python/modem-firmware-1.3+/nrf_cloud_device_mgmt.py

```diff
@@ -361,7 +370,254 @@ def create_fota_job_by_device_id(api_key, bundle_id, update_name, update_desc, d
+                list_out.append(dev.id)
+        else: # updateBy.DEV_ID:
+            list_out.append(dev.id)
+
+    return list_out
+
+def find_or_select_tag(device_list, tag, fota_type):
+    tag_idx = -1
+    tag_list = get_tag_list(device_list)
+
+    if not tag_list:
+            return None
+
+    if tag:
+        try:
+            tag_idx = tag_list.index(tag)
+            if fota_type == updateBundle.fotaType.MODEM:
+                # print warning if tagged devices have different mfw versions installed
+                check_tagged_modem_fw_versions(device_list, tag_list, tag_idx, False)
+        except ValueError:
+            print('Tag list: {}'.format(tag_list))
+            print('Error: No devices found with tag \'{}\''.format(tag))
+            return None
+
+    # ask user to select a tag
+    while tag_idx < 0:
+        tag_idx = user_select_tag(tag_list, device_list)
+        if tag_idx < 0:
+            return None
+
+        if fota_type == updateBundle.fotaType.MODEM:
+            # check the version, if a tag index is returned the user wishes to proceed
+            tag_idx = check_tagged_modem_fw_versions(device_list, tag_list, tag_idx, True)
+
+        # no tag index... try again or exit
+        if tag_idx < 0:
+            print('Select a different tag?')
+            if user_select_yn():
+                continue
+            else:
+                return None
+
+    return tag_list[tag_idx]
+
+def confirm_before_create(name, desc, bundle_id, tag_to_update, update_by):
+    if (name and desc and bundle_id and
+        ((update_by == update_by.TAG and tag_to_update) or (update_by == update_by.DEV_ID))):
+        # if user provides all info via cmd params, do not ask for confirmation
+        print('Creating update with supplied parameters...')
+        return True
+
+    # otherwise, confirm
+    print('Proceed?')
+    return user_select_yn()
+
+def do_job_creation(api_key, bundle_list, device_list, update_by, tag, bundle_id, name, desc, apply, fota_type):
+    filtered_bund_list = []
+    selected_bund_idx = 0
+    job_id = ''
+    tag_to_update = ''
+    cur_fw_ver = ''
+
+    if update_by == updateBy.TAG:
+
+        tag_to_update = find_or_select_tag(device_list, tag, fota_type)
+        if not tag_to_update:
+            return None
+
+    elif update_by == updateBy.DEV_ID:
+
+        cur_fw_ver = print_current_fw_info(device_list[0], fota_type)
+
+    elif update_by == updateBy.BASE_FW_VER:
+
+        if fota_type != updateBundle.fotaType.MODEM:
+            print('Updating by installed FW version is not supported for {} FW'.format(fota_type))
+            return None
+
+        # create a set of the currently installed mfw versions
+        mfw_set = set()
+        for dev in device_list:
+            if dev.mfw_ver:
+                mfw_set.add(dev.mfw_ver)
+            else:
+                print('Warning: device \'{}\' does not have a modem firmware version listed'.format(dev.id))
+
```

Comment:
nit: The newlines in this section are a little wonky

Mostly, remove newlines at the beginning of a newly defined block, even if the following line is the declaration of another block

Reply:
this was on purpose, i found it easier to read with the added new lines.
but i can remove them to be consistent.

Reply:
Readability is subjective, if you think it's more readable and you're doing it on purpose, I'd just go with your gut